### PR TITLE
add default endpoint if in specs

### LINKED ISSUE #395: codegen doesn't respect `host` field in OpenAPI specs
The OpenAPI specifications can optionally include a `host` field, which enables setting the default.

Most often, the specs end up being management.azure.com, however it's not always the case.  For example, the applicationinsights data-plane uses [api.applicationinsights.io](https://github.com/Azure/azure-rest-api-specs/blob/abf7b7e10780d737d99ef0013274431d2024a17d/specification/applicationinsights/data-plane/Microsoft.Insights/preview/v1/AppInsights.json#L18-L19
) and the data-plane of graphrbac uses [graph.windows.net](https://github.com/Azure/azure-rest-api-specs/blob/88e7838a09868a51de3894114355c75929847a46/specification/graphrbac/data-plane/Microsoft.GraphRbac/stable/1.6/graphrbac.json#L8).

Example from the Application Insights spec:

There is another method for setting the default host which should likely get investigated using a Microsoft specific extension called `x-ms-parameterized-host`.  This can be seen in the [eventgrid spec](https://github.com/Azure/azure-rest-api-specs/blob/88e7838a09868a51de3894114355c75929847a46/specification/eventgrid/data-plane/Microsoft.EventGrid/stable/2018-01-01/EventGrid.json#L8-L22)

Right now, the codegen always specifies the default as `management.azure.com`, seen here:

https://github.com/Azure/azure-sdk-for-rust/blob/8cb361b7eca95f01a4f6f88522447b5d8febb771/services/autorust/codegen/src/lib_rs.rs#L69-L78

### PR DESCRIPTION
**Pull request by cataggar**
fix #395

### CODE REVIEW

File: services/svc/imds/src/package_2018_04_02/operations.rs [hunk 1]
```diff
@@ -16,7 +16,7 @@ pub struct ClientBuilder {
     endpoint: Option<String>,
     scopes: Option<Vec<String>>,
 }
-pub const DEFAULT_ENDPOINT: &str = azure_core::resource_manager_endpoint::AZURE_PUBLIC_CLOUD;
+pub const DEFAULT_ENDPOINT: &str = "https://169.254.169.254/metadata";
```
Comment (cataggar) [inline services/svc/imds/src/package_2018_04_02/operations.rs:19]: It is correct. https://docs.microsoft.com/en-us/azure/virtual-machines/windows/instance-metadata-service?tabs=windows

File: services/svc/batch/src/package_2017_06_5_1/operations.rs [hunk 2]
```diff
@@ -16,7 +16,7 @@ pub struct ClientBuilder {
     endpoint: Option<String>,
     scopes: Option<Vec<String>>,
 }
-pub const DEFAULT_ENDPOINT: &str = azure_core::resource_manager_endpoint::AZURE_PUBLIC_CLOUD;
+pub const DEFAULT_ENDPOINT: &str = "https://batch.core.windows.net";
```
Comment (bmc-msft) [inline services/svc/batch/src/package_2017_06_5_1/operations.rs:19]: :thumbsup:

File: services/svc/digitaltwins/src/package_2021_06_30_preview/operations.rs [hunk 3]
```diff
@@ -16,7 +16,7 @@ pub struct ClientBuilder {
     endpoint: Option<String>,
     scopes: Option<Vec<String>>,
 }
-pub const DEFAULT_ENDPOINT: &str = azure_core::resource_manager_endpoint::AZURE_PUBLIC_CLOUD;
+pub const DEFAULT_ENDPOINT: &str = "https://digitaltwins-hostname";
```
Comment (bmc-msft) [inline services/svc/digitaltwins/src/package_2021_06_30_preview/operations.rs:19]: Not needed for this PR, as the previous version was wrong for this crate as well.

1. dotnet: `this.endpoint = endpoint ?? new Uri("https://digitaltwins-name.digitaltwins.azure.net");`
2. python: required to specify as an argument
3. golang: `DefaultBaseURI = "https://digitaltwins-name.digitaltwins.azure.net"`

**Review by bmc-msft — APPROVED** [review]

File: services/svc/digitaltwins/src/package_2021_06_30_preview/operations.rs (hunk 3 above)
Comment (bmc-msft) [reply services/svc/digitaltwins/src/package_2021_06_30_preview/operations.rs:19]: NOTE: As of today (2021-01-10), `digitaltwins-name.digitaltwins.azure.net` doesn't resolve.  That means the dotnet and golang version are broken unless the user specifies an alternate endpoint, which are considered optional in the API.

**Pull request merged**